ui/src/components/ClubsCarousel/index.tsx

```diff
@@ -2,16 +2,40 @@ import styles from './styles.module.scss';
 import Carousel from 'react-multi-carousel';
 import 'react-multi-carousel/lib/styles.css';
 import {useAuthFetch} from '../../api/useAuthFetch';
-import {useEffect} from 'react';
+import {useEffect, useState} from 'react';
 import {useLogout} from '../../hooks/useLogout';
+import {HeadingDropdown} from '../HeadingDropdown';
 
 interface ClubsCarouselData {
   name: string;
   logo: string;
   category: string;
 }
 
+const responsive = {
+  desktop: {
+    breakpoint: {max: 5000, min: 1024},
+    items: 4,
+  },
+  tablet: {
+    breakpoint: {max: 1024, min: 800},
+    items: 2,
+  },
+  mobile: {
+    breakpoint: {max: 800, min: 0},
+    items: 1,
+  },
+};
+
+const options = [
+  {value: 'all', label: 'Clubs'},
+  {value: 'S&T', label: 'Technical Clubs'},
+  {value: 'Cultural', label: 'Cultural Clubs'},
+  {value: 'Sports', label: 'Sports Clubs'},
+];
```

Comment:
Constants should be named in UPPERCASE. [Reference](https://google.github.io/styleguide/tsguide.html#:~:text=Constants%3A%20CONSTANT_CASE%20indicates%20that%20a%20value%20is%20intended%20to%20not%20be%20changed%2C%20and%20may%20be%20used%20for%20values%20that%20can%20technically%20be%20modified%20(i.e.%20values%20that%20are%20not%20deeply%20frozen)%20to%20indicate%20to%20users%20that%20they%20must%20not%20be%20modified.)